sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/reflect/DoFnSignatures.java

```diff
@@ -167,7 +304,10 @@ private static DoFnSignature parseSignature(Class<? extends DoFn> fnClass) {
       Method m,
       TypeToken<?> inputT,
       TypeToken<?> outputT) {
-    errors.checkArgument(void.class.equals(m.getReturnType()), "Must return void");
+    errors.checkArgument(
+        void.class.equals(m.getReturnType())
+            || DoFn.ProcessContinuation.class.equals(m.getReturnType()),
+        "Must return void or ProcessContinuation");
```

Comment:
In general, I'd suggest `"Must return void or %s", ProcessContinuation.class.getSimpleName()` since it is more stable under IDE renamings.